src/commands/period/subcommands/roles/period-roles.handler.ts

```diff
@@ -0,0 +1,48 @@
+import { GuildMember } from "discord.js";
+import { DateTime } from "luxon";
+import { GuildChatInputCommandInteraction } from "../../../../shared/types/GuildChatInputCommandType";
+import { handleChannelAlias } from "../../../../shared/utils/channel-utils";
+import { joinChannel } from "../../../join/join.handler";
+import { PeriodRolesVariables } from "./period-roles.variables";
+
+export const handlePeriodRoles = async (
+	interaction: GuildChatInputCommandInteraction
+): Promise<void> => {
+	const { options } = interaction;
+	await interaction.deferReply({ ephemeral: true });
+	const period = options.getNumber(PeriodRolesVariables.PERIOD, true);
+
+	// Get guild from interaction
+	const guild = interaction.guild;
+	// Iterate through all members in guild
+	await Promise.all(
+		guild.members.cache.map((member) =>
+			updateMember(interaction, member, period)
+		)
+	);
+};
+
+async function updateMember(
+	interaction: GuildChatInputCommandInteraction,
+	member: GuildMember,
+	period: number
+): Promise<void> {
+	const yearString = member?.roles.cache
+		.find((role) => role.name.startsWith("D-"))
+		?.name.slice(2);
+
+	if (!yearString || isNaN(yearString as unknown as number)) return;
+	const year = parseInt(yearString);
```

Comment:
`isNaN(yearString as unknown as number)` can we please not do this?
	`if (!yearString || isNaN(parseInt(yearString))) return;` is prettier imo.

Reply:
the problem with that approach is that parseInt only takes in numbers, not strings. I think you must cast it like this, I might be wrong though